src/software/new_backend/backend.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "software/multithreading/subject.h"
+#include "software/multithreading/threaded_observer.h"
+#include "software/primitive/primitive.h"
+#include "software/world/world.h"
+
+/**
+ * A Backend is an abstraction around communication with robots, vision, and refbox. It
+ * produces SensorMsgs, and consumes primitives that can be sent to the robots.
+ *
+ * This produce/consume pattern is performed by extending both "Observer" and
+ * "Subject". Please see the implementation of those classes for details.
+ */
+class Backend : public Subject<SensorMsg>,
+                public ThreadedObserver<ConstPrimitiveVectorPtr>
+{
+   public:
+    Backend() = default;
+
+    virtual ~Backend() = default;
+
+    // Delete the copy and assignment operators because this class really shouldn't need
+    // them and we don't want to risk doing anything nasty with the internal
+    // multithreading this class potentially uses
+    Backend& operator=(const Backend&) = delete;
+    Backend(const Backend&)            = delete;
+};
```

Comment:
We should definitely have a discussion about this, but I'm not super comfortable with all the `new_` stuff. I'd really prefer that we gradually migrate over the actual backend if possible.

Reply:
The problem is that we're changing the interface of Backend so everything that relies on Backend needs to change as well (unless we create 2 versions of AI, 1 to support sensor fusion, 1 that uses old backend). The other option is to port backends over in one big PR.

Reply:
Agree we should have a talk about this. Faced the same issue with Simulator upgrades.

Reply:
Just quickly comparing this to the old backend, it seems like the main difference is that we're consolidating `Subject<World>` and `Subject<RobotStatus>` into `Subject<SensorMsg>`? I totally get not wanting to have massive PR's though, how does this path sound (each item is  a PR):
- Create Sensor Msg Proto (this PR)
- Update `SensorFusion` to use the Sensor Msg Proto (since `SensorFusion` isn't used by anything else rn iirc)
- Update the backends publish a Sensor Msg Proto (but also still publish `World` and `RobotStatus` for now) 
- Update `full_system_main` to have sensor fusion pull from the backend and have `ai` use `SensorFusion` instead of the `Backend` directly
- Update the backends to not publish `World` or `RobotStatus`

I'm sure I've missed one or two things here, but at this point you'd know better then I would.

Overall I think this is a great opportunity to start getting into incremental updates that keep things neatly integrated, which should help us avoid issues we've had in the past where people go off and develop something and it's a huge pain to integrate later. Totally not saying that's going to happen here (I've got no doubt you'd finish this just fine either way), but it's definitely a habit we should try to get into, both for ourselves and to set examples for others.

Reply:
Makes sense I'll add Subject<SensorMsg> to the current backend.h and stub the other Backends. It's best that we define the interface in a single PR